rule/method/src/main/java/io/opentracing/contrib/specialagent/rule/method/MethodAgentRule.java

```diff
@@ -0,0 +1,74 @@
+package io.opentracing.contrib.specialagent.rule.method;
+
+import io.opentracing.contrib.specialagent.AgentRule;
+import net.bytebuddy.agent.builder.AgentBuilder;
+import net.bytebuddy.asm.Advice;
+import net.bytebuddy.description.type.TypeDescription;
+import net.bytebuddy.dynamic.DynamicType;
+import net.bytebuddy.utility.JavaModule;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static net.bytebuddy.matcher.ElementMatchers.named;
+
+/**
+ * MethodAgentRule
+ *
+ * @author code98@163.com
+ * @date 2019/11/15 5:27 下午
+ */
+public class MethodAgentRule extends AgentRule {
+
+    private static final String METHOD_CONFIG = "sa.config.instrumentation.plugin.method";
+
+    @Override
+    public Iterable<? extends AgentBuilder> buildAgent(AgentBuilder builder) throws Exception {
+        List<AgentBuilder> builders = new ArrayList<>();
+
+        String configInfos = System.getProperty(METHOD_CONFIG);
+        if (configInfos == null || "".equals(configInfos)) {
+            // sa.config.instrumentation.plugin.method is not config
+            return builders;
+        }
```

Comment:
@code98, in case the `configInfos` is null or empty, please return `null`. It is not necessary to instantiate a `new ArrayList<>()`.

Reply:
ok, let me modify the code now

Reply:
![image](https://user-images.githubusercontent.com/10418235/70449959-db720a00-1add-11ea-95b9-b3b60bd23227.png)

renturn `null` it does not work for me, Can only return a `new ArrayList <> ()`

Reply:
This has been fixed [here](https://github.com/opentracing-contrib/java-specialagent/commit/883a9fe4d61bb2f7a348cb31ba6224da4378b1b1).

Reply:
OK，I modify my code now